src/Swift.Bindings/src/Demangler/StringSlice.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using System.Collections.Generic;
+
+namespace BindingsGeneration.Demangling {
+	/// <Summary>
+	/// Represents a a slice which includes facilities similar to StringStream, but
+	/// simplified and more performant. This is build to work specifically with the
+	/// demangler.
+	/// </Summary>
+	internal class StringSlice {
+		string slice;
+
+		/// <Summary>
+		/// Construct a string slice from a string
+		/// </Summary>
+		public StringSlice (string s)
+		{
+			slice = s;
+		}
+
+		/// <Summary>
+		/// Returns true if the slice starts with c
+		/// </Summary>
+		public bool StartsWith (char c)
+		{
+			return Current == c;
+		}
+
+		/// <Summary>
+		/// Returns true if the slice starts with the given string
+		/// </Summary>
+		public bool StartsWith (string s)
+		{
+			if (s == "")
+				return true; // I guess?
```

Comment:
What does this "special" case translate into in real-world scenarios?

Reply:
None other than it seemed to make sense in a mathematical sense and, as it turns out `"foo".StartsWith("")` evaluates to true.